pkg/service/backup/restore.go

```diff
@@ -244,3 +254,29 @@ func (s *Service) forEachRestoredManifest(clusterID uuid.UUID, target RestoreTar
 		return s.forEachManifest(ctx, clusterID, []Location{location}, filter, f)
 	}
 }
+
+// listAllViews is the utility function that queries system_schema.views table to get list of all views created on the cluster.
+func (s *Service) listAllViews(ctx context.Context, clusterID uuid.UUID) (views []string, err error) {
+	// Get cluster session
+	clusterSession, err := s.clusterSession(ctx, clusterID)
+	if err != nil {
+		return nil, errors.Wrap(err, "get CQL cluster session")
+	}
+
+	iter := qb.Select("system_schema.views").
+		Columns("keyspace_name", "view_name").
+		Query(clusterSession).Iter()
+	defer func() {
+		err2 := iter.Close()
+		if err2 != nil {
+			err = multierr.Append(err, err2)
+		}
+	}()
+
+	var keyspace, view string
+	for iter.Scan(&keyspace, &view) {
+		views = append(views, keyspace+"."+view)
+	}
+
+	return
```

Comment:
What is the need for this defer and error appending?
If we get to the point of defer, err always equals nil.
Wouldn't it be nicer to get rid of named return and write:
```go
	var keyspace, view string
	for iter.Scan(&keyspace, &view) {
		views = append(views, keyspace+"."+view)
	}

	return iter.Close()
```